src/installer/tests/TestUtils/TestProjectFixture.cs

```diff
@@ -75,43 +76,45 @@ public void Dispose()
             }
         }
 
+#nullable enable
+
+        private readonly static object? s_buildFilesLock = new object();
+
         private TestProject CopyTestProject(TestProject sourceTestProject)
         {
-            EnsureDirectoryBuildFiles(TestArtifact.TestArtifactsPath);
-            return sourceTestProject.Copy();
-        }
+            lock (s_buildFilesLock)
+            {
+                // Prevent in-process race condition since the TestArtifactsPath is shared by the current
+                // assembly
+                EnsureDirectoryBuildFiles(RepoDirProvider.TestAssetsFolder, TestArtifact.TestArtifactsPath);
```

Comment:
since we are holding the lock, perhaps check if the directory already exists and do nothing? 
Not sure what will happen if we will be creating/writing files while something else (like msbuild?) is reading them.

Reply:
Good point, I was going to skip the file creation, but directoy checking is probably better.

Reply:
Looks like this directory is often used to hold more than just the build files. Conditioning on its existence is going require some significant restructuring.

This should at least address the biggest racing problems.